keys/generate-key-pair.sh

```diff
@@ -0,0 +1,186 @@
+
+if [[ "$1" == "-h" || "$1" == "--help" ]]; then
+    cat << EndOfMessage
+Usage: ${0##*/} [encryption-type]
+
+Arguments:
+  <encryption-type>  The type of encryption to use when generating owner key pair (ecdsa256, ecdsa384, rsa, or all). Will default to all.
+
+Additional environment variables (that do not usually need to be set):
+  SDO_KEY_PWD - The password for your generated keystore. This password must match $containerHome/ocs/config/application.properties/fs.owner.keystore-password
+  KEEP_KEY_FILES - set to 'true' to keep all key pairs generated for each type of key (ecdsa256, ecdsa384, rsa). This is for devs who may want to check out each individual file that goes into generating Key pairs.
+
+EndOfMessage
+    exit 1
+fi
+
+TYPE="${1:-all}"
+CERT=""
+privateKey=""
+KEEP_KEY_FILES=${KEEP_KEY_FILES:-}
+SDO_KEY_PWD=${SDO_KEY_PWD:-}
+
+
+#============================FUNCTIONS=================================
+
+#If the argument passed for this script does not equal one of the encryption types, send error code and exit.
+#BY DEFAULT THE TYPE WILL BE SET TO all
+if [[ -n "$TYPE" ]] && [[ "$TYPE" != "ecdsa256" ]] && [[ "$TYPE" != "ecdsa384" ]] && [[ "$TYPE" != "rsa" ]] && [[ "$TYPE" != "all" ]]; then
+    echo "Error: specified encryption type '$TYPE' is not supported."
+    exit 2
+fi
+
+chk() {
+    local exitCode=$1
+    local task=$2
+    local dontExit=$3   # set to 'continue' to not exit for this error
+    if [[ $exitCode == 0 ]]; then return; fi
+    echo "Error: exit code $exitCode from: $task"
+    if [[ $dontExit != 'continue' ]]; then
+        exit $exitCode
+    fi
+}
+
+ensureWeAreUser() {
+    if [[ $(whoami) = 'root' ]]; then
+        echo "Error: must be normal user to run ${0##*/}"
+        exit 2
+    fi
+}
+
+#This function will create a private key that is needed to create a private keystore. Encryption type passed will decide which command to run for private key creation
+function genKey() {
+#Check if the folder is already created for the type (In case of multiple runs)
+if [[ -d "${TYPE}"Key ]]; then
+    cd "${TYPE}"Key
+else
+    mkdir "${TYPE}"Key && cd "${TYPE}"Key
+fi
+#Generate a private RSA key.
+if [[ $TYPE = "rsa" ]]; then
+    echo -e "Generating a "${TYPE}"private key.\n"
+    openssl genrsa -out "${TYPE}"private-key.pem 2048 > /dev/null 2>&1
+    chk $?
+    certGenerator
+#Generate a private ecdsa (256 or 384) key.
+elif [[ $TYPE = "ecdsa256" ]] || [[ $TYPE = "ecdsa384" ]]; then
+    echo -e "Generating a "${TYPE}"private key.\n"
+    var2=$(echo $TYPE | cut -f2 -da)
+    openssl ecparam -genkey -name secp"${var2}"r1 -out "${TYPE}"private-key.pem > /dev/null 2>&1
+    chk $?
+    certGenerator
+fi
+}
+
+#This function will create a key certificate that is needed to create a corresponding public key, as well as a private keystore.
+function certGenerator() {
+    if [[ -f "${TYPE}"private-key.pem ]]; then
+        privateKey=""${TYPE}"private-key.pem"
```

Comment:
Variables in functions should be defined with the `local` keyword, because by default all variables have global scope, which you should avoid.